platform/m6/lib.c

```diff
@@ -0,0 +1,142 @@
+#include "platform.h"
+#include "lolevel.h"
+#include "live_view.h"
+
+#include "lib.h"
+
+void shutdown_soft() {
+	_PostLogicalEventForNotPowerType(0x1005, 0);
+}
+
+/**
+ * @see main startup
+ */
+void shutdown() {
+	while (1)
+		_SleepTask(500);
+}
```

Comment:
what should I do here ?  how do I determine the magic number 0x1005 ?

Reply:
That comes from the firmware's levent (logical event) table. It's the event named "PressOffButton". Your camera doesn't have such a button, you'll need an event that does the same on your hardware, such as "PowerLeverOff".
BTW, that shutdown() implementation is defunct, it just traps the task executing it.

Reply:
so I could use this lua script to find all the event names when I got the alt mode working... https://chdk.fandom.com/wiki/LogicalEvent

Reply:
You can do this step later when you're nearly finished porting as both of these are only called from scripts.
There's also a CHDK tool for dumping the levent table, tools/find_levent[.c], it only gets built if you issue _make extras_ in the tools dir.